x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateDoubleGroupingAggregatorFunction.java

```diff
@@ -568,8 +581,8 @@ public void evaluateFinal(Block[] blocks, int offset, IntVector selected, Groupi
                 for (int i = 1; i < intervals.length; i++) {
                     Interval next = intervals[i - 1]; // reversed
                     Interval prev = intervals[i];
-                    if (prev.v2 > next.v2) {
-                        state.resets += prev.v2;
+                    if (prev.v1 > next.v2) {
+                        state.resets += prev.v1;
```

Comment:
This change caught the bug - I will split the rate tests across multiple indices to cover this case.